File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:

        low = 0
        res =float('inf')
        sum =0

        for high in range(len(nums)):

            sum+=nums[high]

            while sum >= target:

                length = high -low+1
                res = min(res , length)

                sum -= nums[low]
                low+=1  

        return 0 if res == float('inf') else res             

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {

        int n = nums.length;
        int low = 0;
        int high = 0;
        int subarray_sum = 0;
        int res = Integer.MAX_VALUE;

        while(high < n)
        {
            subarray_sum+=nums[high];

            while(subarray_sum >= target)
            {
                int len  = high-low+1;
                res = Math.min(res, len);

                subarray_sum -= nums[low];
                low++;
            }
            high++;
        }
        
        return (res == Integer.MAX_VALUE) ? 0:res;
    }
}